src/Compilers/Core/Portable/PEWriter/MetadataWriter.cs

```diff
@@ -2715,7 +2715,31 @@ private void PopulateTypeDefTableRows()
             foreach (INamedTypeDefinition typeDef in typeDefs)
             {
                 INamespaceTypeDefinition namespaceType = typeDef.AsNamespaceTypeDefinition(Context);
-                string mangledTypeName = GetMangledName(typeDef);
+
+                var moduleBuilder = Context.Module;
+                int generation = 0;
+
+                if (moduleBuilder.PreviousGeneration != null)
```

Comment:
>if (moduleBuilder.PreviousGeneration != null) [](http://example.com/codeflow?start=16&length=45)

Typically we've used `DeltaMetadataWriter` for EnC changes. Could this code be moved to `DeltaMetadataWriter`?

Reply:
Yes, we can make GetMangledName virtual

Reply:
Moved the code to `CommonPEModuleBuilder.GetTypeDefinitionGeneration` - it uses abstract properties of the builder anyways.